golem-worker-service-base/src/gateway_execution/to_response.rs

```diff
@@ -12,82 +12,157 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use super::auth_call_back_binding_handler::{AuthorisationError, AuthorisationSuccess};
+use super::file_server_binding_handler::FileServerBindingSuccess;
+use super::http_handler_binding_handler::{HttpHandlerBindingError, HttpHandlerBindingSuccess};
+use super::RibInputTypeMismatch;
 use crate::api::WorkerApiBaseError;
-use crate::gateway_binding::HttpRequestDetails;
-use crate::gateway_execution::auth_call_back_binding_handler::AuthCallBackResult;
-use crate::gateway_execution::file_server_binding_handler::{
-    FileServerBindingError, FileServerBindingResult,
-};
+use crate::gateway_execution::file_server_binding_handler::FileServerBindingError;
 use crate::gateway_execution::gateway_session::GatewaySessionStore;
+use crate::gateway_execution::request::RichRequest;
 use crate::gateway_execution::to_response_failure::ToHttpResponseFromSafeDisplay;
 use crate::gateway_middleware::HttpCors as CorsPreflight;
+use crate::gateway_rib_interpreter::EvaluationError;
 use async_trait::async_trait;
 use http::header::*;
 use http::StatusCode;
 use poem::Body;
 use poem::IntoResponse;
 use rib::RibResult;
 
-use super::http_handler_binding_handler::{HttpHandlerBindingError, HttpHandlerBindingResult};
-
 #[async_trait]
 pub trait ToHttpResponse {
```

Comment:
I probably want to replace this trait with a single function that works on GatewayHttpResult later, but for now leaving like this